airbyte-webapp/src/pages/connections/CreateConnectionPage/usePreloadData.ts

```diff
@@ -0,0 +1,113 @@
+import { useEffect } from "react";
+import { useLocation, useNavigate, useSearchParams } from "react-router-dom";
+
+import {
+  DestinationDefinitionRead,
+  DestinationRead,
+  SourceDefinitionRead,
+  SourceRead,
+} from "core/request/AirbyteClient";
+import { useGetDestination } from "hooks/services/useDestinationHook";
+import { useGetSource } from "hooks/services/useSourceHook";
+import { useDestinationDefinition } from "services/connector/DestinationDefinitionService";
+import { useSourceDefinition } from "services/connector/SourceDefinitionService";
+
+export function hasSourceId(state: unknown): state is { sourceId: string } {
+  return typeof state === "object" && state !== null && typeof (state as { sourceId?: string }).sourceId === "string";
+}
+
+export function hasDestinationId(state: unknown): state is { destinationId: string } {
+  return (
+    typeof state === "object" &&
+    state !== null &&
+    typeof (state as { destinationId?: string }).destinationId === "string"
```

Comment:
Same idea here. Could also type as `state: { destinationId?: string }`

Reply:
see [the comment above](https://github.com/airbytehq/airbyte/pull/22701#discussion_r1107581261)